camel/benchmarks/apibench.py

```diff
@@ -0,0 +1,342 @@
+        programming framework used.\n4. $EXPLANATION should be \
+        a step-by-step explanation.\n5. The $CODE is the python code.\n6. \
+        Do not repeat the format in your answer."
+    )
+    return prompt
+
+
+class APIBenchBenchmark(BaseBenchmark):
+    r"""APIBench Benchmark adopted from `Gorilla: Large Language Model
+    Connected with Massive APIs`
+    <https://huggingface.co/datasets/gorilla-llm/APIBench>.
+
+    Args:
+        data_dir (str): The directory to save the data.
+        save_to (str): The file to save the results.
+        processes (int, optional): The number of processes to use.
+            (default: :obj:`1`)
+    """
+
+    # TODO: Integrate retriever (pending)
+
+    def __init__(
+        self,
+        data_dir: str,
+        save_to: str,
+        processes: int = 1,
+    ):
+        r"""Initialize the APIBench benchmark.
+
+        Args:
+            data_dir (str): The directory to save the data.
+            save_to (str): The file to save the results.
+            processes (int, optional): The number of processes to use for
+                parallel processing. (default: :obj:`1`)
+        """
+        super().__init__("apibench", data_dir, save_to, processes)
+
+    def download(self):
+        r"""Download the APIBench dataset."""
+        from huggingface_hub import snapshot_download
+
+        snapshot_download(
+            repo_id="gorilla-llm/APIBench",
+            repo_type="dataset",
+            local_dir=self.data_dir,
+            local_dir_use_symlinks=True,
+        )
+
+        def download_github_subdirectory(
+            repo, subdir, branch="main", data_dir=self.data_dir
+        ):
+            r"""Download subdirectory of the Github repo of the benchmark."""
+            api_url = f"https://api.github.com/repos/{repo}/contents/{subdir}?ref={branch}"
+            headers = {"Accept": "application/vnd.github.v3+json"}
+            response = requests.get(api_url, headers=headers)
+            response.raise_for_status()
+            files = response.json()
+            os.makedirs(data_dir, exist_ok=True)
+
+            for file in tqdm(files, desc="Downloading"):
+                file_path = data_dir / file["name"]
+
+                if file["type"] == "file":
+                    file_url = file["download_url"]
+                    file_response = requests.get(file_url)
+                    with open(file_path, "wb") as f:
+                        f.write(file_response.content)
+                elif file["type"] == "dir":
+                    download_github_subdirectory(
+                        repo,
+                        os.path.join(subdir, file["name"]),
+                        branch,
+                        file_path,
+                    )
+
+        repo = "ShishirPatil/gorilla"
+        subdir = "eval/eval-data/questions"
+
+        download_github_subdirectory(repo, subdir)
+
+    def load(self, dataset_name):
```

Comment:
The `load` method is missing a type annotation for `dataset_name`.

Reply:
Hi Xukun I've removed the dataset_name arg and set a new attribute for level/dataset_name configuration.